### PR TITLE
Prepare for 15.0.0 Release

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4468

# Rationale for this change

# What changes are included in this PR?

- Increase version number
- Updated CHANGELOG ([rendered view](https://github.com/andygrove/arrow-datafusion/blob/prep-15.0.0/datafusion/CHANGELOG.md))

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: test-utils/Cargo.toml [hunk 1]
```diff
@@ -24,6 +24,6 @@ edition = "2021"
 
 [dependencies]
 arrow = { version = "28.0.0", features = ["prettyprint"] }
-datafusion-common = { path = "../datafusion/common", version = "14.0.0" }
```
Comment (andygrove) [inline test-utils/Cargo.toml:27]: No need to specify the version here since we do not release the `test-utils` crate

Comment (avantgardnerio) [reply test-utils/Cargo.toml:27]: I think technically there's no need for these redundant version numbers anywhere now that I fixed the docker script.

**Review by avantgardnerio — APPROVED** [review]

**Comment by andygrove** [comment]
@avantgardnerio the version numbers are required by `cargo publish`. We can only publish crates that depend on other published crates.

**Comment by alamb** [comment]
🆗  looks good -- thanks @andygrove

**Pull request merged**